lib/Entity/Schedule.php

```diff
@@ -531,9 +531,21 @@ public function validate()
             // additional validation for Interrupt Layout event type
             if ($this->eventTypeId == Schedule::$INTERRUPT_EVENT) {
 
-                if (!v::intType()->notEmpty()->validate($this->shareOfVoice) || !v::min(0)->validate($this->shareOfVoice)
-                    || !v::max(100)->validate($this->shareOfVoice)) {
-                    throw new InvalidArgumentException(__('Share of Voice must be a whole number between 0 and 100'), 'shareOfVoice');
+                // Hack : If this is an interrupt, check that the column is a SMALLINT and if it isn't alter the table
+                $sql = 'SELECT DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE table_name = :table_name AND COLUMN_NAME = :column_name';
+                $params = ['table_name' => 'schedule', 'column_name' => 'shareOfVoice' ];
+                $results = $this->store->select($sql, $params);
+
+                if (count($results) > 0) {
+                    $dataType = $results[0]['DATA_TYPE'];
+
+                    if ($dataType !== 'smallint') {
+                        $this->store->update('ALTER TABLE `schedule` MODIFY `shareOfVoice` SMALLINT', []);
```

Comment:
Looking at this I had an idea - could we use this detection as an opportunity to convert any existing interrupt schedules?

We could say ```UPDATE `schedule` SET shareOfVoice = 3600 * (shareOfVoice / 100) WHERE shareOfVoice > 0```

I think that would convert any existing percentages to their correct seconds/hour? For example 10% share of voice would be 3600 * (10/100) = 360 ? What do you think?